elements/lisk-codec/src/types.ts

```diff
@@ -21,13 +21,21 @@ export interface GenericObject {
 export interface SchemaPair {
 	readonly [key: string]: SchemaProps;
 }
+
 export interface Schema {
+	readonly $id: string;
+	readonly type: string;
+	readonly properties: object;
+}
+
+export interface ValidatedSchema {
 	readonly $id: string;
 	readonly $schema?: string;
```

Comment:
Logically validated schema must have `$schema`, otherwise its can't be validated against the lisk meta schema and fall back to Draft-07 of JSON-Schema, which will allow any type to be a root level type.